.travis.yml

```diff
@@ -19,7 +19,7 @@ matrix:
   - name: cargo build --no-default-features
     script:
       - cargo build --no-default-features --verbose
-      - cargo build --all --verbose
+      - cargo build --all --all-targets --verbose
```

Comment:
As far as I can see there are just examples and tests which will both be checked by `cargo test --all`, why does this need `--all-targets`? (I think this separate `cargo build --all` in the `--no-default-features` build should just be removed, it's not really part of `--no-default-features`).

Reply:
(If you want to stick `--all-targets` somewhere having it in the clippy build would be great to ensure idiomatic example code, but that's likely to need some actual fixes so would be best done as a separate PR).

Reply:
Commit removed from PR.